### PR TITLE
Add recipe for pysmb

### PR DESCRIPTION
**Pull request by rzillman**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/pysmb/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.zip
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
```
Comment (scopatz) [inline recipes/pysmb/meta.yaml:15]: This can be `noarch: python`

Comment (rzillman) [reply recipes/pysmb/meta.yaml:16]: Thanks @scopatz

File: recipes/pysmb/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    - python
+    - pip
+  run:
+    - python
+    - pyasn1
+
+test:
+  imports:
+    - smb
+    - nmb
```
Comment (marcelotrevisani) [inline recipes/pysmb/meta.yaml:29]: Can we try to add ``pip check`` here please?
```suggestion
  requires:
    - pip
  commands:
    - pip check
```

File: recipes/pysmb/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+about:
+  home: https://github.com/miketeo/pysmb
+  license: ZLIB
```
Comment (marcelotrevisani) [inline recipes/pysmb/meta.yaml:33]:
```suggestion
  license: Zlib
```

File: recipes/pysmb/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  license_family: OTHER
```
Comment (marcelotrevisani) [inline recipes/pysmb/meta.yaml:34]:
```suggestion
  license_family: Zlib
```

Comment (rzillman) [reply recipes/pysmb/meta.yaml:34]: This is not supported for `conda-build` v3.19.2
```
about/license_family 'Zlib' not allowed. Allowed families are AGPL,
APACHE, BSD, CC, GPL, GPL2, GPL3, LGPL, MIT, MOZILLA, NONE, OTHER,
PROPRIETARY, PSF, and PUBLIC-DOMAIN.
```

File: recipes/pysmb/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  license_file: LICENSE
+  summary: pysmb is an experimental SMB/CIFS library written in Python to support file sharing between Windows and Linux machine
+  doc_url: https://github.com/miketeo/pysmb
+  dev_url: https://github.com/miketeo/pysmb
+
+extra:
+  recipe-maintainers:
+    - dhirschfeld
```
Comment (dhirschfeld) [inline recipes/pysmb/meta.yaml:46]: :+1: happy to help maintain this recipe

**Comment by dhirschfeld** [comment]
I think this is good to merge?

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**